src/game/AI/BaseAI/GuardAI.cpp

```diff
@@ -138,6 +138,6 @@ void GuardAI::AttackStart(Unit* u)
 
 void GuardAI::JustDied(Unit* killer)
 {
-    if (Player* pkiller = killer->GetCharmerOrOwnerPlayerOrPlayerItself())
+    if (Player* pkiller = killer->GetBeneficiaryPlayer())
```

Comment:
Since its usage mostly lies within XP/loot/honor rewards, I couldn't find a better term to describe the unit which behaves on behalf of another in this most common context.